### PR TITLE
[AIC-py] cli rage part 2: issue draft

### PR DESCRIPTION
**Pull request by jonathanlastmileai**
[AIC-py] cli rage part 2: issue draft

Try to do a lot of stuff automatically. Run it and you'll see exactly what I mean :)

`aiconfig rage`

Screenshot 2024-01-12 at 1 33 48 PM

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/903).
* __->__ #903
* #870
* #869

### CODE REVIEW

File: python/src/aiconfig/scripts/rage/rage.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    server_log_str = f"see {server_log_path}" if server_log_path else "Couldn't find the log file :( Check your terminal output"
+    aiconfig_log_str = f"see {aiconfig_log_path}" if aiconfig_log_path else "Couldn't find the log file :( Check your terminal output"
+
+    commands_to_run = [
+        "which pip",
+        "which pip3",
+        "which python",
+        "which python3",
+        "python --version",
+        "python3 --version",
```
Comment (rholinshead) [inline python/src/aiconfig/scripts/rage/rage.py:86]: Would we want to also include the python-aiconfig package version here, or can that be derived elsewhere?

File: python/src/aiconfig/scripts/rage/rage.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        f"""
+        This is a rage-generated bug report. Like an ordinary bug report, but with more rage.
+
+        * With markdown
+        * Cool, huh?
+
+        [TODO: detailed description]
+        [TODO: steps to reproduce (what were you doing, what did you expect to happen, what happened instead?)]
+        [TODO: server log contents ({server_log_str})]
+        [TODO: aiconfig log contents ({aiconfig_log_str})]]
```
Comment (rholinshead) [inline python/src/aiconfig/scripts/rage/rage.py:100]: nit, can we make these more clear for user action?
```suggestion
        [add detailed description here]
        [provide steps to reproduce (what were you doing, what did you expect to happen, what happened instead?)]
        [paste server log contents ({server_log_str})]
        [paste aiconfig log contents ({aiconfig_log_str})]]
```

File: python/src/aiconfig/scripts/rage/rage.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    print(f"Title: {title}")
+    print(f"Body:\n{body}")
+    open_draft = _get_yes_or_no_input("Would you like to open a draft issue in your browser? [Y/n] ")
+    if open_draft:
+        _troll_the_user_part_2()
+        _open_github_issue_draft(
+            repo="lastmile-ai/aiconfig",
+            title=title,
+            body=body,
+            labels=["type: bug", "status: planned"],
```
Comment (rholinshead) [inline python/src/aiconfig/scripts/rage/rage.py:121]: should they be 'planned' before we've looked at them? Ideally there would be a 'needs triage' bucket to put them into

File: python/src/aiconfig/scripts/rage/rage.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    commands_to_run = [
+        "which pip",
+        "which pip3",
+        "which python",
+        "which python3",
+        "python --version",
+        "python3 --version",
+    ]
+    command_inputs_outputs = dedent("\n\n".join(_fmt_command_input_output(command, _try_run_command(command)) for command in commands_to_run))
+    pip_list_output = _get_pip_list_filtered()
```
Comment (rholinshead) [inline python/src/aiconfig/scripts/rage/rage.py:89]: Oh, I guess it's here, nvm

File: python/src/aiconfig/scripts/rage/rage.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+
+    assert type in ["spinner", "music"]
+
+    spinning = itertools.cycle(["-", "/", "|", "\\"])
+
+    def get_animation():
+        if type == "spinner":
+            return next(spinning)
+        else:
+            return "".join(np.random.choice(["♩", "♫", "♬", "♪"]) for _ in range(5))
```
Comment (rholinshead) [inline python/src/aiconfig/scripts/rage/rage.py:174]: Eagerly awaiting the r-rex jumping over cacti in part 3

**Review by rholinshead — APPROVED** [review]

**Comment by rholinshead** [comment]
Accepting to unblock, just minor things

File: python/src/aiconfig/scripts/rage/rage.py (hunk 3 above)
Comment (jonathanlastmileai) [reply python/src/aiconfig/scripts/rage/rage.py:121]: I don't think this exists, didnt want to make a new tag.

File: python/src/aiconfig/scripts/rage/rage.py (hunk 4 above)
Comment (jonathanlastmileai) [reply python/src/aiconfig/scripts/rage/rage.py:89]: I special cased it because of the filtering step.

**Pull request merged**